src/node_crypto.cc

```diff
@@ -2623,6 +2642,89 @@ void SSLWrap<Base>::GetCipher(const FunctionCallbackInfo<Value>& args) {
 }
 
 
+template <class Base>
+void SSLWrap<Base>::GetSharedSigalgs(const FunctionCallbackInfo<Value>& args) {
+  Base* w;
+  ASSIGN_OR_RETURN_UNWRAP(&w, args.Holder());
+  Environment* env = w->ssl_env();
+  Local<Context> context = env->context();
+  Local<Array> ret_arr = Array::New(env->isolate());
+
+  SSL* ssl = w->ssl_.get();
+  int nsig = SSL_get_shared_sigalgs(ssl, 0, nullptr, nullptr, nullptr, nullptr,
+                                    nullptr);
+
+  for (int i = 0; i < nsig; i++) {
+    int hash_nid;
+    int sign_nid;
+    std::string sig_with_md;
+
+    SSL_get_shared_sigalgs(ssl, i, &sign_nid, &hash_nid, nullptr, nullptr,
+                           nullptr);
+
+    switch (sign_nid) {
+      case EVP_PKEY_RSA:
+        sig_with_md = std::string("RSA+");
+        break;
+
+      case EVP_PKEY_RSA_PSS:
+        sig_with_md = std::string("RSA-PSS+");
+        break;
+
+      case EVP_PKEY_DSA:
+        sig_with_md = std::string("DSA+");
+        break;
+
+      case EVP_PKEY_EC:
+        sig_with_md = std::string("ECDSA+");
+        break;
+
+      case NID_ED25519:
+        sig_with_md = std::string("Ed25519+");
+        break;
+
+      case NID_ED448:
+        sig_with_md = std::string("Ed448+");
+        break;
+
+      case NID_id_GostR3410_2001:
+        sig_with_md = std::string("gost2001+");
+        break;
+
+      case NID_id_GostR3410_2012_256:
+        sig_with_md = std::string("gost2012_256+");
+        break;
+
+      case NID_id_GostR3410_2012_512:
+        sig_with_md = std::string("gost2012_512+");
+        break;
+
+      default:
+        const char* sn = OBJ_nid2sn(sign_nid);
+
+        if (sn != nullptr) {
+          sig_with_md = std::string(sn) + "+";
+        } else {
+          sig_with_md = std::string("UNDEF+");
```

Comment:
not sure why this is using `std::string()` to initialize here, but not on line 2717 below. I'd remove this, and the ones above, if they aren't required.

Reply:
Still required on 2706 because of the addition though.